lightning/src/ln/outbound_payment.rs

```diff
@@ -2834,9 +2832,28 @@ mod tests {
 		assert!(pending_events.lock().unwrap().is_empty());
 	}
 
-	fn invoice_request() -> InvoiceRequest {
-		let invreq_bytes = <Vec<u8>>::from_hex("00200101010101010101010101010101010101010101010101010101010101010101080203e80a00162102bb58b5feca505c74edc000d8282fc556e51a1024fc8e7d7e56c6f887c5c8d5f25821035be5e9478209674a96e60f1f037f6176540fd001fa1d64694770c56a7709c42cf040a31d68198578a5aa1fe57de5f6d33f89c1556752cb333dbb56ac727f751893804d3a26b42909f1952fb1db8080238f9476b829160387692df35ff85dfe72e2ed").unwrap();
-		Readable::read(&mut Cursor::new(&invreq_bytes[..])).unwrap()
+	fn dummy_invoice_request() -> InvoiceRequest {
+		let secp_ctx = Secp256k1::new();
+		let recipient_pubkey = {
+			let secret_key = SecretKey::from_slice(&<Vec<u8>>::from_hex("4141414141414141414141414141414141414141414141414141414141414141").unwrap()).unwrap();
+			Keypair::from_secret_key(&secp_ctx, &secret_key).public_key()
+		};
+		let payer_keys = {
+			let secret_key = SecretKey::from_slice(&<Vec<u8>>::from_hex("4242424242424242424242424242424242424242424242424242424242424242").unwrap()).unwrap();
+			Keypair::from_secret_key(&secp_ctx, &secret_key)
+		};
+		OfferBuilder::new(recipient_pubkey)
+			.description("test".into())
+			.amount_msats(1000)
+			.build()
+			.unwrap()
+			.request_invoice(vec![0; 8], payer_keys.public_key()).unwrap()
+			.build()
+			.unwrap()
+			.sign(|message: &UnsignedInvoiceRequest|
+				Ok(secp_ctx.sign_schnorr_no_aux_rand(message.as_ref().as_digest(), &payer_keys))
+			)
```

Comment:
This is interface being removed in #3264. Could use the derived version and `build_and_sign` instead?